translation-web/src/main/java/eu/europeana/api/translation/web/service/LangDetectionWebService.java

```diff
@@ -29,21 +32,24 @@ public class LangDetectionWebService extends BaseWebService {
 
   public LangDetectResponse detectLang(LangDetectRequest langDetectRequest)
       throws EuropeanaI18nApiException {
+    List<LanguageDetectionObj> languageDetectionObjs = buildLangDetectionObjectList(langDetectRequest);
+
     LanguageDetectionService langDetectService = getLangDetectService(langDetectRequest);
     LanguageDetectionService fallback = getFallbackService(langDetectRequest);
     List<String> langs = null;
     String serviceId = null;
     try {
-      langs =
-          langDetectService.detectLang(langDetectRequest.getText(), langDetectRequest.getLang());
+      // preprocess the values
+      translationServiceProvider.getLanguageDetectionPreProcessor().detectLang(languageDetectionObjs);
+      langDetectService.detectLang(languageDetectionObjs.stream().filter(to -> to.isTranslatable()).collect(Collectors.toList()));
       serviceId = langDetectService.getServiceId();
     } catch (LanguageDetectionException originalError) {
       // check if fallback is available
       if (fallback == null) {
         throwApiException(originalError);
       } else {
         try {
-          langs = fallback.detectLang(langDetectRequest.getText(), langDetectRequest.getLang());
+          fallback.detectLang(languageDetectionObjs.stream().filter(to -> to.isTranslatable()).collect(Collectors.toList()));
```

Comment:
we should remove the code dupplication by building only once the list of translatable texts (see line 44)